launchable/utils/tracking.py

```diff
@@ -0,0 +1,63 @@
+from enum import Enum
+from typing import Dict, Any, Optional, Union
+
+from launchable.utils.http_client import LaunchableClient
+from launchable.version import __version__
+
+
+class Tracking:
+    # General events
+    class Event(Enum):
+        SHALLOW_CLONE = 'shallow_clone'  # this event is an example
+
+    # Error events
+    class ExceptionEvent(Enum):
+        UNKNOWN_ERROR = 'UNKNOWN_ERROR'
+        INTERNAL_CLI_ERROR = 'INTERNAL_CLI_ERROR'
+        # Errors related to requests package
+        NETWORK_ERROR = 'NETWORK_ERROR'
+        TIMEOUT_ERROR = 'TIMEOUT_ERROR'
+        INTERNAL_ERROR = 'INTERNAL_ERROR'
+        UNEXPECTED_HTTP_STATUS_ERROR = 'UNEXPECTED_HTTP_STATUS_ERROR'
+
+    class Command(Enum):
+        VERIFY = 'VERIFY'
+        RECORD_TESTS = 'RECORD_TESTS'
+        RECORD_BUILD = 'RECORD_BUILD'
+        RECORD_SESSION = 'RECORD_SESSION'
+        SUBSET = 'SUBSET'
+
+
+class TrackingClient:
+    def __init__(self, http_client: LaunchableClient) -> None:
+        self.http_client = http_client
+
+    def send_error_event(
+        self,
+        command: Tracking.Command,
+        event_name: Union[Tracking.Event, Tracking.ExceptionEvent],
+        stack_trace: str,
+        organization: str = "",
+        workspace: str = "",
+        api: str = "",
+        metadata: Optional[Dict[str, Any]] = None
+    ):
+        if metadata is None:
+            metadata = {}
+        metadata["stackTrace"] = stack_trace
+        if organization:
+            metadata["organization"] = organization
+        if workspace:
+            metadata["workspace"] = workspace
+        if api:
+            metadata["api"] = api
```

Comment:
Don't check null or not on server-side? if we already checked it, do we need to check the values?

Can we use the below, can't we?

```python
if metadata is None:
    metadata = {
       stackTrace: stack_trace,
       organization: organization,
       workspace: workspace,
       api: api
    }
```

Reply:
[a97d623](https://github.com/launchableinc/cli/pull/594/commits/a97d62380533c4f7e532500703dc824768d70a16)